### PR TITLE
Limit concurrent web requests

### PR DESCRIPTION
**Pull request by eanders-MS**
This is a quick fix to stabilize AltQuiz. Globally limit the number of concurrent asset fetch web requests in the MRE runtime.

### CODE REVIEW

File: MREUnityRuntime/MREUnityRuntimeLib/Util/Unity/AssetFetcher.cs [hunk 1]
```diff
@@ -67,13 +81,16 @@ IEnumerator LoadCoroutine()
...
 						}
 						else if (typeof(T) == typeof(Texture))
 						{
-							result.Asset = ((DownloadHandlerTexture) handler).texture as T;
+							result.Asset = ((DownloadHandlerTexture)handler).texture as T;
 						}
 					}
+
+					// Decrement activeLoads
+					Interlocked.Decrement(ref activeLoads);
```
Comment (eanders-MS) [inline MREUnityRuntime/MREUnityRuntimeLib/Util/Unity/AssetFetcher.cs:93]: Would be nice to be able to decrement the counter before the finalization handler was called.

File: MREUnityRuntime/MREUnityRuntimeLib/Assets/AssetLoader.cs [hunk 2]
```diff
@@ -146,125 +146,153 @@ private async Task LoadAssets(LoadAssets payload, Action onCompleteCallback)
 			onCompleteCallback?.Invoke();
 		}
 
+		// Global: Keep track of the number of active glTF web requests.
+		private static int activeLoads = 0;
+
 		private async Task<IList<Asset>> LoadAssetsFromGLTF(AssetSource source, Guid containerId, ColliderType colliderType)
 		{
+			// Spin asynchronously until activeLoads allows us through.
+			while (activeLoads >= 4)
```
Comment (StLoch) [inline MREUnityRuntime/MREUnityRuntimeLib/Assets/AssetLoader.cs:155]: wbn to have the '4' as a shared constant for easy tuning / readability.

Comment (eanders-MS) [reply MREUnityRuntime/MREUnityRuntimeLib/Assets/AssetLoader.cs]: Good call. Updated.

File: MREUnityRuntime/MREUnityRuntimeLib/Assets/AssetLoader.cs [hunk 3]
```diff
@@ -148,123 +148,135 @@ private async Task LoadAssets(LoadAssets payload, Action onCompleteCallback)
...
 
-			GLTFSceneImporter importer =
-				MREAPI.AppsAPI.GLTFImporterFactory.CreateImporter(gltfRoot, loader, _asyncHelper, stream);
-			importer.SceneParent = MREAPI.AppsAPI.AssetCache.CacheRootGO().transform;
-			importer.Collider = colliderType.ToGLTFColliderType();
-
-			// load textures
-			if (gltfRoot.Textures != null)
+			using (GLTFSceneImporter importer =
+				MREAPI.AppsAPI.GLTFImporterFactory.CreateImporter(gltfRoot, loader, _asyncHelper, stream))
```
Comment (stevenvergenz) [inline MREUnityRuntime/MREUnityRuntimeLib/Assets/AssetLoader.cs:187]: Oh good catch, I didn't realize glTF loaders were disposable. Are we sure disposing this doesn't destroy the in-memory assets that are in-use?

Comment (eanders-MS) [reply MREUnityRuntime/MREUnityRuntimeLib/Assets/AssetLoader.cs:187]: I haven't changed the lifetime of the loader. It was already being disposed, but the call was explicit (line 266 on the left-hand side). This is just a refinement to make the Dispose call implicit, but also makes the lifetime tolerant against exceptions.

**Pull request merged**